CSharpSourceCode/Abilities/Crosshairs/SingleTargetCrosshair.cs

```diff
@@ -25,24 +25,25 @@ public override void Hide()
 
         private void FindTarget()
         {
-            Vec3 startRay, endRay;
-            //startRay = _missionScreen.CombatCamera.Frame.origin;
-            //endRay = _missionScreen.CombatCamera.Frame.Elevate(_template.MaxDistance).origin;
-            _missionScreen.ScreenPointToWorldRay(new Vec2(960,540), out startRay, out endRay);
-            var newTarget = Mission.Current.RayCastForClosestAgent(startRay, endRay, out _, _caster.Index, 0.01f);
+            Vec2 mousePositionRanged = Input.MousePositionRanged;
+            Vec3 sourcePoint;
+            Vec3 targetPoint;
+            _missionScreen.ScreenPointToWorldRay(mousePositionRanged, out sourcePoint, out targetPoint);
```

Comment:
does this work? it would be awesome if it did, we couldnt get it to work when trying :( The ray was never correct.

Reply:
yes, it works, but i found that it's fucked up since you press zoom on horseback. I know where this issue from but don't know how to fix it properly. Will look for solution

Reply:
fixed